src/main/java/org/primefaces/component/datatable/DataTable.java

```diff
@@ -1325,6 +1338,7 @@ public Object saveState(FacesContext context) {
         }
 
         setSortByAsMap(null);
+        setFilterByAsMap(null);
```

Comment:
Will this have the same regression issue as sorting?

Reply:
I don't think so, because each filters have their own input so it will always be submitted to the server (which is not the case for sorting). But since, filterMeta should live over multiple request I'm planning to remove this yes

Reply:
You might be right actually, I don't have any use case in mind, but even if inputs are passed to the request, you still need the "encodingFeature" parameter to enable filtering. Anyway, don't worry I'll remove this :wink: